contrib/guix/libexec/build-osx.sh

```diff
@@ -0,0 +1,262 @@
+# GCC_LIB="${CROSS_GCC_LIBS[0]}" # ...we just want the first one (there should only be one)
+
+# # Determine output paths to use in CROSS_* environment variables
+# CROSS_GLIBC="$(store_path glibc-cross-${HOST})"
+# CROSS_GLIBC_STATIC="$(store_path glibc-cross-${HOST} static)"
+# CROSS_KERNEL="$(store_path linux-libre-headers-cross-${HOST})"
+# CROSS_GCC="$(store_path gcc-cross-${HOST})"
+# CROSS_GCC_LIBS=( "${CROSS_GCC}/lib/gcc/${HOST}"/* ) # This expands to an array of directories...
+# CROSS_GCC_LIB="${CROSS_GCC_LIBS[0]}" # ...we just want the first one (there should only be one)
+
+# # Set environment variables to point Guix's cross-toolchain to the right
+# # includes/libs for $HOST
+# #
+# # NOTE: CROSS_C_INCLUDE_PATH is missing ${CROSS_GCC_LIB}/include-fixed, because
+# # the limits.h in it is missing a '#include_next <limits.h>'
+# #
+# export CROSS_C_INCLUDE_PATH="${CROSS_GCC_LIB}/include:${CROSS_GLIBC}/include:${CROSS_KERNEL}/include"
+# export CROSS_CPLUS_INCLUDE_PATH="${CROSS_GCC}/include/c++:${CROSS_GCC}/include/c++/${HOST}:${CROSS_GCC}/include/c++/backward:${CROSS_C_INCLUDE_PATH}"
+# export CROSS_LIBRARY_PATH="${CROSS_GCC}/lib:${CROSS_GCC}/${HOST}/lib:${CROSS_GCC_LIB}:${CROSS_GLIBC}/lib:${CROSS_GLIBC_STATIC}/lib"
+
+# Correct
+export C_INCLUDE_PATH="${GLIBC}/include:${KERNEL}/include"
+export CPLUS_INCLUDE_PATH="${GCC}/include/c++:${GCC}/include/c++/x86_64-unknown-linux-gnu:${GCC}/include/c++/backward:${C_INCLUDE_PATH}"
+export OBJC_INCLUDE_PATH="${C_INCLUDE_PATH}"
+export CPATH="$(store_path util-linux)/include:$(store_path libcap)/include:$(store_path zlib)/include:$(store_path bzip2)/include"
+# export LIBRARY_PATH="${CROSS_GCC}/lib:${CROSS_GCC}/${HOST}/lib:${CROSS_GCC_LIB}:${CROSS_GLIBC}/lib:${CROSS_GLIBC_STATIC}/lib"
+
+
+# IFS=':' read -ra PATHS <<< "${C_INCLUDE_PATH}:${CPLUS_INCLUDE_PATH}:${CPATH}"
+# for p in "${PATHS[@]}"; do
+#     if [ ! -d "$p" ]; then
+#         echo "'$p' doesn't exist or isn't a directory... Aborting..."
+#         exit 1
+#     fi
+# done
+
+# Disable Guix ld auto-rpath behavior
+# export GUIX_LD_WRAPPER_DISABLE_RPATH=yes
+
+# Make /usr/bin if it doesn't exist
+[ -e /usr/bin ] || mkdir -p /usr/bin
+
+# Symlink file and env to a conventional path
+[ -e /usr/bin/file ] || ln -s --no-dereference "$(command -v file)" /usr/bin/file
+[ -e /usr/bin/env ]  || ln -s --no-dereference "$(command -v env)"  /usr/bin/env
+
+# Determine the correct value for -Wl,--dynamic-linker for the current $HOST
+# glibc_dynamic_linker=$(
+#     case "$HOST" in
+#         i686-linux-gnu)      echo /lib/ld-linux.so.2 ;;
+#         x86_64-linux-gnu)    echo /lib64/ld-linux-x86-64.so.2 ;;
+#         arm-linux-gnueabihf) echo /lib/ld-linux-armhf.so.3 ;;
+#         aarch64-linux-gnu)   echo /lib/ld-linux-aarch64.so.1 ;;
+#         riscv64-linux-gnu)   echo /lib/ld-linux-riscv64-lp64d.so.1 ;;
+#         *)                   exit 1 ;;
+#     esac
+# )
+
+# Environment variables for determinism
```

Comment:
I assume we no longer have to set `ZERO_AR_DATE` here because we're not using a libtool from cctools?

From [libtool.c](https://opensource.apple.com/source/cctools/cctools-927.0.2/misc/libtool.c.auto.html):
> all of these values are initialized to reasonable defaults for deterministic
archives: archives that are consistent regardless of user, time, or umask
differences. Ordinarily, these values and will be set to more specific
ones when building new archives, but that can be suppressed using the
'-D' option or the ZERO_AR_DATE environment variable.